.github/scripts/therock_configure_ci.py

```
@@ -25,8 +29,45 @@ def set_github_output(d: Mapping[str, str]):
     with open(step_output_file, "a") as f:
         f.writelines(f"{k}={v}" + "\n" for k, v in d.items())
         
+def get_modified_paths(base_ref: str) -> Optional[Iterable[str]]:
+    # """Returns the paths of modified files relative to the base reference."""
+    try:
+        return subprocess.run(
+            ["git", "diff", "--name-only", base_ref],
+            stdout=subprocess.PIPE,
+            check=True,
+            text=True,
+            timeout=60,
+        ).stdout.splitlines()
+    except TimeoutError:
+        print(
+            "Computing modified files timed out. Not using PR diff to determine"
+            " jobs to run.",
+            file=sys.stderr,
+        )
+        return None
```

Should this "fail open" or "fail closed"? If a timeout is reached, I would expect lots of files to be modified (or some other error), in which case I think it would be safer to run more jobs instead of less jobs.

---

probably fail open! seems worse to fail if there's some random error

I think best idea is to add retry and attempt it a few times with backoff and worse case, just return empty

---

Retrying a few times sounds good. I think in the case that getting the list of files fails, we should run all CI jobs, not no CI jobs.

That change can be made separately from this PR though.

---

sgtm